### ISSUE TITLE
Source address dropdown incomplete on Outbound NAT rules

### ISSUE THREAD

**Issue by monreal** [opnsense/core#2837]
After switching Outbound NAT to manual on 18.7.5 I tried to re-create the automatic rules. I found that the "Source address" dropdown is missing the networks list (automatic rule includes "LAN networks").

**Comment by fichtner** [comment]
LAN networks is a macro inside pf(4) only used for automatic rule generation.

Why do you emulate automatic with manual? Hybrid mode not what you're looking for?

**Comment by monreal** [comment]
The automatic rule does not include the "static-port" option, which I need. I read in multiple tutorials that I need to duplicate the rules manually but none actually said how to do it. Am I missing something simple? Is there another way to set "static-port"?

**Comment by fichtner** [comment]
No, you're right. Not sure how tricky this is to add. @AdSchellevis what do you think?

**Comment by AdSchellevis** [comment]
@monreal @fichtner just to be sure, only the network portion in the source address dropdown? That shouldn't be too hard, I can probably fix that for the next version.

**Comment by fichtner** [comment]
Yes, the same mechanic that automatic rules are using :)

**Comment by AdSchellevis** [comment]
This https://github.com/opnsense/core/commit/9be373ff759efbebb0af8056d603c832bbbe4c9d should do the trick. rule parsing is generic indeed, no changes needed there.

**Comment by monreal** [comment]
> @monreal @fichtner just to be sure, only the network portion in the source address dropdown?

Well, after looking at this again it looks like the "Networks" list is missing from the "Destination address" selection, too. Seems to be the same problem.

Then I also noticed a second problem in the same dialog, which however is probably not related: I get an error when saving if I select "WAN address" for "Translation / target". I can only save if I use "Interface address". Should I open another issue for this?

**Comment by AdSchellevis** [comment]
@monreal what's the error?

**Comment by monreal** [comment]
It says:

The following input errors were detected:
A valid target IP address must be specified.

**Comment by AdSchellevis** [comment]
I can't seem to replicate this on my end, can you post some screenshots of the settings?

**Comment by monreal** [comment]
![1](https://user-images.githubusercontent.com/54005/47524665-83f1ea00-d89b-11e8-9ada-d1654a174cf1.png)
![2](https://user-images.githubusercontent.com/54005/47524669-88b69e00-d89b-11e8-9643-63ee34daf528.png)

**Comment by monreal** [comment]
![3](https://user-images.githubusercontent.com/54005/47524688-95d38d00-d89b-11e8-942d-08f790c6fdcf.png)

**Comment by monreal** [comment]
"LAN_networks" is an alias for "192.168.178.0/24 127.0.0.0/8" btw, which is what the automatic rule contains

**Comment by AdSchellevis** [comment]
I can't replicate this, this message can't appear when selecting "Interface address", since targetip is empty in that case:

Empty assigned here:
https://github.com/opnsense/core/blob/9be373ff759efbebb0af8056d603c832bbbe4c9d/src/www/firewall_nat_out_edit.php#L654

Output when not valid...
https://github.com/opnsense/core/blob/9be373ff759efbebb0af8056d603c832bbbe4c9d/src/www/firewall_nat_out_edit.php#L191-L193


I can look at the destination field, can't help you with this, maybe some weird browser issue on your end.

**Comment by monreal** [comment]
> I can look at the destination field, can't help you with this, maybe 
> some weird browser issue on your end.

Big thanks anyway! Let's keep this issue about fixing the dropdowns. Would be nice if you could fix the destination address dropdown in addition to the source address dropdown. After that I guess this issue can be closed as resolved.

**Comment by AdSchellevis** [comment]
@monreal ok, I'll close it with the commit, feel free to reopen if you experience issues with the network dropdowns.
